src/context-selection/data-set-selector-bar-item/use-selectable-data-sets.js

```diff
@@ -0,0 +1,29 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+
+const QUERY_SELECTABLE_DATA_SETS = {
+    dataSets: {
+        resource: 'dataSets',
+        params: {
+            paging: false,
+        },
+    },
+}
+
+export default function useSelectableDataSets() {
+    const { loading, error, data } = useDataQuery(QUERY_SELECTABLE_DATA_SETS)
+
+    // @TODO: How to handle pages / large lists?
```

Comment:
If we're going to go in the direction I'm suspecting, we could use this: https://react-query.tanstack.com/guides/paginated-queries

Reply:
This is about whether we want to handle pagination at all.
Maybe large lists is something that's rather unusual in the wild and therefore we can omit pagination.